android/app/src/main/assets/js/core.js

```diff
@@ -465,7 +465,10 @@ window.addEventListener('DOMContentLoaded', async () => {
     }
 
     if (reader.generalSettings.val.removeExtraParagraphSpacing) {
-      html = html.replace(/<\s*br[^>]*>/gi, '\n').replace(/\n{2,}/g, '\n\n');
+      html = html
+        .replace(/([\u200B-\u200D\uFEFF]|&nbsp;)(?<=<p>\1)/g, '')
+        .replace(/(?:<br>\s*<\/?br>)+/g, '')
```

Comment:
why adding `+` in the end. Isnt `g` flag is enough

Reply:
it's faster

Reply:
You should be the one who provide a benchmark :).
But yeah, it maybe true

Reply:
https://github.com/user-attachments/assets/84dea670-a2b7-4e11-86e6-4da85ad99d16